client/src/components/Connections/components/Search.js

```diff
@@ -0,0 +1,91 @@
+// @flow
+import React from 'react'
+import {withRouter} from 'react-router-dom'
+import {withHandlers, withState} from 'recompose'
+import {compose} from 'redux'
+import {connect} from 'react-redux'
+import {updateValue} from '../../../actions/sharedActions'
+import {Form, FormGroup, Label, Input, Button} from 'reactstrap'
+import type {ContextRouter} from 'react-router'
+
+import EntitySearchWrapper, {type EntitySearchProps} from '../dataWrappers/EntitySearchWrapper'
+import './Search.css'
+
+type EmptyHandler = () => void // TODO extract
+type EventHandler<T> = (e: T) => void
+
+const checkEnter = (callback: EmptyHandler): EventHandler<KeyboardEvent> => {
```

Comment:
Why is this a function returning function? As far as I can see, it's called always with the same callback and those props are causing rerender on each check (since the reference returned from `checkEnter` is always different).